spec/features/hq/crud_sponsor_spec.rb

```diff
@@ -0,0 +1,190 @@
+    fill_in 'Name', :with => 'Sponsor1'
+    select 'Active', :from => 'Status'
+    select 'Male', :from => 'Gender'
+    fill_in 'Start date', :with => '2014-08-25'
+    fill_in 'Requested orphan count', :with => '2'
+    select 'Individual', :from => 'sponsor_sponsor_type_id'
+    select Branch.all.first.name, :from => 'sponsor_branch_id'
+    select '(Afghanistan) أفغانستان', :from => 'Country'
+    select Sponsor.all_cities.first, :from => 'City'
+    select @sponsors.first.agent.user_name, :from => 'sponsor_agent_id'
+    click_button 'Create Sponsor'
+    and_i_should_be_on :hq_sponsor_path, { sponsor_name: 'Sponsor1' }
+    and_i_should_see 'Sponsor successfuly saved'
+    and_i_should_see 'Sponsor1'
+  end
+
+  scenario 'Should be able to edit a sponsor from the sponsor show page' do
+    name = @sponsors.first.name
+    given_i_am_on_page_for_sponsor 'Show Sponsor', name
+    click_link 'Edit Sponsor'
+    and_i_should_be_on :edit_hq_sponsor_path, { sponsor_name: name }
+    fill_in 'Additional info', :with => 'Additional Information'
+    click_button 'Update Sponsor'
+    and_i_should_be_on :hq_sponsor_path, { sponsor_name: name }
+    and_i_should_see 'Sponsor successfuly saved'
+    and_i_should_see 'Additional Information'
+  end
+
+  scenario "can't edit sponsor_type, organization or branch" do
+    given_i_am_on_page_for_sponsor 'Edit Sponsor', @sponsors.first.name
+    i_should_not_be_able_to_change 'Sponsor type', 'sponsor'
+    i_should_not_be_able_to_change 'Organization', 'sponsor'
+    i_should_not_be_able_to_change 'Branch', 'sponsor'
+  end
+
+  scenario "can't delete a sponsor from the sponsor show page" do
+    given_i_am_on_page_for_sponsor 'Show Sponsor', @sponsors.first.name
+    and_i_should_not_see 'Delete Sponsor'
+  end
+
+  scenario "returns to partner show page on cancelling edit partner" do
+    name = @sponsors.first.name
+    given_i_am_on_page_for_sponsor 'Show Sponsor', name
+    click_link 'Edit Sponsor'
+    fill_in 'Additional info', :with => 'Additional Information'
+    click_link 'Cancel'
+    and_i_should_be_on :hq_sponsor_path, { sponsor_name: name }
+    and_i_should_not_see 'Additional Information'
+  end
+
+  private
+
+  def i_should_see_the_following_fields_on_the_page table
+    table.each do |hash|
+      expect(page).to have_content(hash[:value])
+    end
+  end
+
+  def given_i_am_on_page_for_sponsor page_name, sponsor_name = nil
+    if sponsor_name.nil?
+      visit path_to_sponsor_role page_name
+    else
+      sponsor = Sponsor.find_by name: sponsor_name unless sponsor_name.nil?
+      visit path_to_sponsor_role page_name, sponsor.id
+    end
+  end
+
+  def path_to_sponsor_role page_name, id = nil
+    name = page_name.downcase
+    case name
+      when 'edit sponsor' then
+        edit_hq_sponsor_path id
+      when 'show sponsor' then
+        hq_sponsor_path id
+      when 'new sponsor' then
+        new_hq_sponsor_path
+    end
+  end
```

Comment:
If we move the expectation from `get_path` to the end of `and_i_should_be_on` it will do what the method says.
And you can use `get_path` instead of the new `path_to_sponsor_role` method, like you specified. 

This also implies checking all scenarios that use `get_path` and make sure an expectation is not necessary there (if expectation is necessary then it probably needs `and_i_should_be_on`).

Yes we should use only `page` in `get_path`.

Yes, that `field_hash.keys.first.to_sym` is brutal.
Maybe something like:
```
case field_hash.keys.map(&:to_sym)
  when [:sponsor_name] ...
  when [:sponsor_name, :user_name] ... #in future
```